### ISSUE TITLE
Click Publish in CMS Editor becomes read only

### ISSUE THREAD

**Issue by RVXD** [Firesphere/silverstripe-solr-search#140]
**Describe the bug**
If you add Solr indexing to SiteTree and use the CMS to edit pages. After you made a change to the page and hit Publish button the page becomes uneditable. You have to reload the CMS for it to work again. All action buttons in the bottom are gone.

**Normal situation**
![Screenshot 2019-11-26 at 09 39 26](https://user-images.githubusercontent.com/1586761/69613957-b50a9400-1032-11ea-920a-a10fd7348592.png)

**After click on publish**
![Screenshot 2019-11-26 at 09 41 09](https://user-images.githubusercontent.com/1586761/69614002-c0f65600-1032-11ea-9917-1fc9d0e31ca6.png)

**Caused by**
This is caused by SolrDataObjectExtension method onAfterPublish.
I used Injector to replace the SolrDataObjectExtension with my own version that doesn't trigger this is action. We use the indexing task periodically to update the indexes.

```
class SolrDataObjectExtension extends DataObjectExtension
{
    public function onAfterPublish()
    {
        return; // original onAfterPublish causes issues in CMS, page is not editible after publish
    }
}
```

**To Reproduce**
1. Install clean SilverStripe
2. Add Solr module
3. Create Simple Index
4. dev/build and open CMS
5. Make changes to the page and click on publish button

**My composer file:**

```
{
    "name": "silverstripe/installer",
    "type": "silverstripe-recipe",
    "description": "The SilverStripe Framework Installer",
    "require": {
        "php": ">=5.6.0",
        "silverstripe/recipe-plugin": "^1.2",
        "silverstripe/recipe-cms": "4.4.4@stable",
        "silverstripe-themes/simple": "~3.2.0",
        "firesphere/solr-search": "^0.8.3"
    },
    "require-dev": {
        "phpunit/phpunit": "^5.7"
    },
    "extra": {
        "resources-dir": "_resources",
        "project-files-installed": [
            "app/.htaccess",
            "app/_config.php",
            "app/_config/mysite.yml",
            "app/src/Page.php",
            "app/src/PageController.php"
        ],
        "public-files-installed": [
            ".htaccess",
            "index.php",
            "install-frameworkmissing.html",
            "install.php",
            "web.config"
        ]
    },
    "config": {
        "process-timeout": 600
    },
    "prefer-stable": true,
    "minimum-stability": "dev"
}
```
**My Solr Index file:**
```
class SiteTreeIndex extends BaseIndex
{
    public function init()
    {
        $this->addClass(SiteTree::class);
        $this->addFulltextField('Title',null,['boost'=>2]);
        $this->addFulltextField('Content');
    }

    /**
     * @return string
     * installed solr via:
     * https://computingforgeeks.com/install-latest-apache-solr-on-ubuntu-debian/
     */

    public function getIndexName()
    {
        return 'zandbak';
    }
}
```
**Solr version**
Solr 8.2.0 is working normally.

![Screenshot 2019-11-26 at 10 00 35](https://user-images.githubusercontent.com/1586761/69614626-aec8e780-1033-11ea-8cab-15a15ac4be65.png)

![Screenshot 2019-11-26 at 10 02 39](https://user-images.githubusercontent.com/1586761/69614750-f2235600-1033-11ea-98ac-50b7998d6dda.png)

**Comment by Firesphere** [comment]
Nice catch, I had seen this behaviour but never thought/realised it would be related.

**Comment by Firesphere** [comment]
Resolved in PR #141 , please continue discussion there.
